### PR TITLE
Add retry check for Sidekiq

### PR DESCRIPTION
**Pull request by loed-idzinga**
Hi there! 

In our company we are also checking the retry queue if any of the jobs go over the `DEFAULT_RETRY_CHECK` limit (for Sidekiq it defaults to 25), so we can take a look to them manually. Thought maybe I could share the code?

Our approach was to create a custom provider, so if this approach is okay I'll continue and write some specs.

### CODE REVIEW

**Comment by lbeder** [comment]
This is great! Could you just make sure that every rubocop style issues are fixed (see here: https://github.com/lbeder/health-monitor-rails/actions/runs/3853885912/jobs/6578427525)?

**Comment by loed-idzinga** [comment]
@lbeder I've updated the code 👍

**Comment by lbeder** [comment]
Thanks, @loed-idzinga! We're almost there. There are still very minor rubocop offenses (see here: https://github.com/lbeder/health-monitor-rails/actions/runs/3910525936/jobs/6692671069) and I've left you a comment in the PR itself.

File: spec/support/providers.rb [hunk 1]
```diff
@@ -80,4 +84,10 @@ def stub_sidekiq_queue_size_failure(queue_name = 'default')
   def stub_sidekiq_redis_failure
     allow(Sidekiq).to receive(:redis_info).and_raise(Redis::CannotConnectError)
   end
+
+  def stub_sidekiq_over_retry_limit_failure
+    retry_set = instance_double(Sidekiq::RetrySet)
+    allow(retry_set).to receive(:select).and_return(["item" => { retry_count: 21 }])
```
Comment (lbeder) [inline spec/support/providers.rb:90]: Perhaps something like `::HealthMonitor::Providers::Sidekiq::Configuration::DEFAULT_RETRY_CHECK + 1` instead?

Comment (loed-idzinga) [reply spec/support/providers.rb]: Good catch! 👍

**Comment by loed-idzinga** [comment]
@lbeder Thanks for the review, I've updated the code with the suggestions. 

I've moved the `let(:..)` statements around to fix `RSpec/MultipleMemoizedHelpers` Rubocop cop. Let me know if you agree with this change.

**Review by lbeder — APPROVED** [review]

**Pull request merged**